New pacman-git breaks yay

### Affected Version
<!-- Please ensure you are using the latest yay-git package
Use `yay -V` to get installed version
Example: `yay v8.1139.r0.g9ac4ab6 - libalpm v11.0.1` -->
10.2.1.r7.g29f4c43

### Describe the bug
<!-- A clear and concise description of the bug. -->
`yay-git` can't build with latest `pacman-git`
Version of `pacman-git` is `6.0.0alpha1.r99.g558d08d8`
ps. Old version `6.0.0alpha1.r86.gc294b7cb` doesn't have this issue

### Reproduction Steps
1. Install latest `pacman-git`
2. Try build `yay-git`

### Expected behavior
<!-- A clear and concise description of what you expected to happen. -->
`yay-git` should build.

### Output
<!-- 
Include the FULL output of any relevant commands/configs
The current yay config can be printed with `yay -Pg`
Paste services are only needed for excessive output (>500 lines)
-->
```sh
go build -trimpath -mod=readonly -modcacherw -tags six -ldflags '-X "main.yayVersion=10.2.1.r7.g29f4c43" -X "main.localePath=/usr/share/locale/" -linkmode=external' -buildmode=pie -o yay
go: downloading github.com/Jguer/go-alpm v0.0.0-20200405152916-a3feea4322e9
go: downloading github.com/Morganamilo/go-pacmanconf v0.0.0-20180910220353-9c5265e1b14f
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/mikkeloscar/aur v0.0.0-20200113170522-1cb4e2949656
go: downloading github.com/leonelquinteros/gotext v1.4.0
go: downloading github.com/Morganamilo/go-srcinfo v1.0.0
go: downloading golang.org/x/term v0.0.0-20210220032956-6a3ed077a48d
go: downloading github.com/Jguer/go-alpm/v2 v2.0.2
go: downloading github.com/stretchr/testify v1.7.0
go: downloading golang.org/x/sys v0.0.0-20210309074719-68d13333faf2
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
# github.com/Jguer/yay/v10
/usr/lib/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/usr/bin/ld: /tmp/go-link-491984825/000007.o: in function `_cgo_c94cb4692741_Cfunc_alpm_option_get_arch':
handle.cgo2.c:(.text+0x235): undefined reference to `alpm_option_get_arch'
/usr/bin/ld: /tmp/go-link-491984825/000007.o: in function `_cgo_c94cb4692741_Cfunc_alpm_option_set_arch':
handle.cgo2.c:(.text+0x7c9): undefined reference to `alpm_option_set_arch'

```

--- a/go.mod
+++ b/go.mod
@@ -2,8 +2,8 @@ module github.com/Jguer/yay/v10
 
 require (
 	github.com/Jguer/go-alpm v0.0.0-20200405152916-a3feea4322e9 // indirect
-	github.com/Jguer/go-alpm/v2 v2.0.2
-	github.com/Morganamilo/go-pacmanconf v0.0.0-20180910220353-9c5265e1b14f
+	github.com/Jguer/go-alpm/v2 v2.0.3-0.20210503120603-1e8b5ec6f821
+	github.com/Morganamilo/go-pacmanconf v0.0.0-20210502114700-cff030e927a5
 	github.com/Morganamilo/go-srcinfo v1.0.0
 	github.com/bradleyjkemp/cupaloy v2.3.0+incompatible
 	github.com/davecgh/go-spew v1.1.1 // indirect

--- a/install.go
+++ b/install.go
@@ -443,10 +443,24 @@ func earlyRefresh(cmdArgs *settings.Arguments) error {
 	return config.Runtime.CmdRunner.Show(passToPacman(arguments))
 }
 
+func alpmArchIsSupported(alpmArch []string, arch string) bool {
+	if arch == "any" {
+		return true
+	}
+
+	for _, a := range alpmArch {
+		if a == arch {
+			return true
+		}
+	}
+
+	return false
+}
+
 func getIncompatible(bases []dep.Base, srcinfos map[string]*gosrc.Srcinfo, dbExecutor db.Executor) (stringset.StringSet, error) {
 	incompatible := make(stringset.StringSet)
 	basesMap := make(map[string]dep.Base)
-	alpmArch, err := dbExecutor.AlpmArch()
+	alpmArch, err := dbExecutor.AlpmArchitectures()
 	if err != nil {
 		return nil, err
 	}
@@ -454,7 +468,7 @@ func getIncompatible(bases []dep.Base, srcinfos map[string]*gosrc.Srcinfo, dbExe
 nextpkg:
 	for _, base := range bases {
 		for _, arch := range srcinfos[base.Pkgbase()].Arch {
-			if arch == "any" || arch == alpmArch {
+			if alpmArchIsSupported(alpmArch, arch) {
 				continue nextpkg
 			}
 		}

--- a/main.go
+++ b/main.go
@@ -47,8 +47,8 @@ func initAlpm(cmdArgs *settings.Arguments, pacmanConfigPath string) (*pacmanconf
 		pacmanConf.DBPath = dbPath
 	}
 
-	if arch, _, exists := cmdArgs.GetArg("arch"); exists {
-		pacmanConf.Architecture = arch
+	if arch := cmdArgs.GetArgs("arch"); arch != nil {
+		pacmanConf.Architecture = append(pacmanConf.Architecture, arch...)
 	}
 
 	if ignoreArray := cmdArgs.GetArgs("ignore"); ignoreArray != nil {

--- a/main_test.go
+++ b/main_test.go
@@ -20,7 +20,7 @@ func TestPacmanConf(t *testing.T) {
 		HoldPkg:                []string(nil),
 		IgnorePkg:              []string{"ignore", "this", "package"},
 		IgnoreGroup:            []string{"ignore", "this", "group"},
-		Architecture:           "8086",
+		Architecture:           []string{"8086"},
 		XferCommand:            "",
 		NoUpgrade:              []string{"noupgrade"},
 		NoExtract:              []string{"noextract"},

--- a/pkg/db/executor.go
+++ b/pkg/db/executor.go
@@ -22,7 +22,7 @@ type Upgrade struct {
 }
 
 type Executor interface {
-	AlpmArch() (string, error)
+	AlpmArchitectures() ([]string, error)
 	BiggestPackages() []IPackage
 	Cleanup()
 	IsCorrectVersionInstalled(string, string) bool

--- a/pkg/db/ialpm/alpm.go
+++ b/pkg/db/ialpm/alpm.go
@@ -115,7 +115,7 @@ func configureAlpm(pacmanConf *pacmanconf.Config, alpmHandle *alpm.Handle) error
 		return err
 	}
 
-	if err := alpmHandle.SetArch(pacmanConf.Architecture); err != nil {
+	if err := alpmSetArchitecture(alpmHandle, pacmanConf.Architecture); err != nil {
 		return err
 	}
 
@@ -440,10 +440,6 @@ func (ae *AlpmExecutor) RepoUpgrades(enableDowngrade bool) ([]db.Upgrade, error)
 	})
 
 	return slice, nil
-}
-
-func (ae *AlpmExecutor) AlpmArch() (string, error) {
-	return ae.handle.Arch()
 }
 
 func (ae *AlpmExecutor) BiggestPackages() []alpm.IPackage {

--- a/pkg/db/ialpm/alpm_five.go
+++ b/pkg/db/ialpm/alpm_five.go
@@ -1,0 +1,17 @@
+// +build !six
+
+package ialpm
+
+import (
+	alpm "github.com/Jguer/go-alpm/v2"
+)
+
+func alpmSetArchitecture(alpmHandle *alpm.Handle, arch []string) error {
+	return alpmHandle.SetArch(arch[0])
+}
+
+func (ae *AlpmExecutor) AlpmArchitectures() ([]string, error) {
+	arch, err := ae.handle.Arch()
+
+	return []string{arch}, err
+}

--- a/pkg/db/ialpm/alpm_five_test.go
+++ b/pkg/db/ialpm/alpm_five_test.go
@@ -1,0 +1,13 @@
+// +build !six
+
+package ialpm
+
+import (
+	alpm "github.com/Jguer/go-alpm/v2"
+)
+
+func alpmTestGetArch(h *alpm.Handle) ([]string, error) {
+	arch, err := h.Arch()
+
+	return []string{arch}, err
+}

--- a/pkg/db/ialpm/alpm_six.go
+++ b/pkg/db/ialpm/alpm_six.go
@@ -1,0 +1,17 @@
+// +build six
+
+package ialpm
+
+import (
+	alpm "github.com/Jguer/go-alpm/v2"
+)
+
+func alpmSetArchitecture(alpmHandle *alpm.Handle, arch []string) error {
+	return alpmHandle.SetArchitectures(arch)
+}
+
+func (ae *AlpmExecutor) AlpmArchitectures() ([]string, error) {
+	architectures, err := ae.handle.GetArchitectures()
+
+	return architectures.Slice(), err
+}

--- a/pkg/db/ialpm/alpm_six_test.go
+++ b/pkg/db/ialpm/alpm_six_test.go
@@ -1,0 +1,13 @@
+// +build six
+
+package ialpm
+
+import (
+	alpm "github.com/Jguer/go-alpm/v2"
+)
+
+func alpmTestGetArch(h *alpm.Handle) ([]string, error) {
+	architectures, err := h.GetArchitectures()
+
+	return architectures.Slice(), err
+}

--- a/pkg/db/ialpm/alpm_test.go
+++ b/pkg/db/ialpm/alpm_test.go
@@ -18,7 +18,7 @@ func TestAlpmExecutor(t *testing.T) {
 		HoldPkg:                []string(nil),
 		IgnorePkg:              []string{"ignore", "this", "package"},
 		IgnoreGroup:            []string{"ignore", "this", "group"},
-		Architecture:           "8086",
+		Architecture:           []string{"8086"},
 		XferCommand:            "",
 		NoUpgrade:              []string{"noupgrade"},
 		NoExtract:              []string{"noextract"},
@@ -75,9 +75,9 @@ func TestAlpmExecutor(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, []string{"/usr/share/libalpm/hooks/", "/hookdir/"}, hook.Slice())
 
-	arch, err := h.Arch()
+	arch, err := alpmTestGetArch(h)
 	assert.Nil(t, err)
-	assert.Equal(t, "8086", arch)
+	assert.Equal(t, []string{"8086"}, arch)
 
 	ignorePkg, err := h.IgnorePkgs()
 	assert.Nil(t, err)

